holoviews/core/util.py

```diff
@@ -710,6 +710,18 @@ def max_range(ranges):
         return (np.NaN, np.NaN)
 
 
+def dimension_range(lower, upper, dimension):
+    """
+    Computes the range along a dimension by combining the data range
+    with the Dimension soft_range and range.
+    """
```

Comment:
I would be surprised if we don't have very similar logic to this somewhere else in holoviews already...

Reply:
I see you replaced two chunks of code with this utility below, is that everywhere this utility could be used?